pkg/kubectl/cmd/create_role.go

```diff
@@ -336,6 +338,12 @@ func generateResourcePolicyRules(mapper meta.RESTMapper, verbs []string, resourc
 		rule.ResourceNames = resourceNames
 		rules = append(rules, rule)
 	}
+	for _, u := range nonResourceURLs {
+		rule := rbac.PolicyRule{}
+		rule.Verbs = verbs
```

Comment:
verbs are going to be tricky if you are trying to create a role containing both resources and non-resource urls (the verbs that apply to those are different... see https://kubernetes.io/docs/admin/authorization/#review-your-request-attributes)